Mathlib/NumberTheory/SiegelsLemma.lean

```diff
@@ -0,0 +1,215 @@
+  rw [card_T_eq, card_S_eq]
+  rify --this is necessary because ‖A‖ is a real number
+  calc
+  ∏ x : Fin m, (∑ x_1 : Fin n, ↑B * ↑(A x x_1)⁺ - ∑ x_1 : Fin n, ↑B * -↑(A x x_1)⁻ + 1)
+    ≤ (n * ‖A‖ * B + 1) ^ m := by
+      rw [← Fin.prod_const m (n * ‖A‖ * B + 1)] --transform ^m into a product
+      apply Finset.prod_le_prod --two goals
+      all_goals intro i _
+      · have h := N_le_P_add_one m n A i
+        rify at h
+        linarith only [h]
+      · simp only [mul_neg, sum_neg_distrib, sub_neg_eq_add, add_le_add_iff_right]
+        have h1 : n * ‖A‖ * B = ∑ _ : Fin n, ‖A‖ * B := by
+          simp only [sum_const, card_univ, Fintype.card_fin, nsmul_eq_mul, mul_assoc ↑n ‖A‖ ↑B]
+        rw [h1, ← Finset.sum_add_distrib]
+        gcongr with j _
+        rw [← mul_add, ← Int.cast_add, posPart_add_negPart (A i j), mul_comm]
+        apply mul_le_mul_of_nonneg_right _ (Nat.cast_nonneg B)
+        rw [Int.cast_abs]
+        exact norm_entry_le_entrywise_sup_norm A
+  _  ≤ (n * ‖A‖) ^ m * (B + 1) ^ m := by
+        rw [← mul_pow]
+        apply pow_le_pow_left (add_nonneg _ (zero_le_one' ℝ)) _ --two goals
+        · exact mul_nonneg (mul_nonneg (Nat.cast_nonneg n) (norm_nonneg A)) (Nat.cast_nonneg B)
+        · rw [mul_add]
+          simp only [mul_one, add_le_add_iff_left]
+          exact Left.one_le_mul_of_le_of_le
+              (by exact_mod_cast Nat.one_le_of_lt hn) (one_le_norm_A_of_ne_zero m n A hA)
+              (Nat.cast_nonneg n)
+  _ = ((n * ‖A‖) ^ (m / ((n : ℝ) - m))) ^ ((n : ℝ) - m)  * (B + 1) ^ m := by
+        rw [mul_left_inj' (pow_ne_zero m (Nat.cast_add_one_ne_zero B)),
+          ← rpow_mul (mul_nonneg (Nat.cast_nonneg n) (norm_nonneg A)), ← Real.rpow_natCast,
+          div_mul_cancel₀]
+        exact sub_ne_zero_of_ne (mod_cast hn.ne')
+  _ < (B + 1) ^ ((n : ℝ) - m) * (B + 1) ^ m := by
+        rw [mul_lt_mul_right (pow_pos (Nat.cast_add_one_pos B) m)]
+        apply rpow_lt_rpow --three goals
+        · exact rpow_nonneg (mul_nonneg (Nat.cast_nonneg n) (norm_nonneg A)) e
+        · exact Nat.lt_floor_add_one ((n * ‖A‖) ^ e)
+        · simpa only [sub_pos, Nat.cast_lt] using hn
+  _ = (B + 1) ^ n := by
+        rw [← rpow_natCast, ← rpow_add (Nat.cast_add_one_pos B), ← rpow_natCast, sub_add_cancel]
+
+end preparation
+
+theorem exists_ne_zero_int_vec_norm_le  (hA_nezero : A ≠ 0) : ∃ t : Fin n → ℤ, t ≠ 0 ∧
+    A *ᵥ t = 0 ∧ ‖t‖ ≤ ((n * ‖A‖) ^ ((m : ℝ) / (n - m))) := by
+  --Pigeonhole
+  rcases Finset.exists_ne_map_eq_of_card_lt_of_maps_to
+    (card_S_lt_card_T m n A hn hA_nezero) (image_T_subset_S m n A)
+    with ⟨x, hxT, y, hyT, hneq, hfeq⟩
+  --proofs that x - y ≠ 0 and x-y is a solution
+  refine ⟨x - y, sub_ne_zero.mpr hneq, by simp only [mulVec_sub, sub_eq_zero, hfeq], ?_⟩
+  --Inequality
+  have n_mul_norm_A_pow_e_nonneg : 0 ≤ (n * ‖A‖) ^ e := by positivity
+  have : ‖x - y‖ = ‖col (Fin 1) (x - y)‖ := by exact Eq.symm (norm_col (x - y))
+  rw [this, norm_le_iff n_mul_norm_A_pow_e_nonneg]
```

Comment:
Earlier today, when I updated Mathlib, the proof broke here.
Instead of `this` there used to be `← norm_col` but it gave the error 
`typeclass instance problem is stuck, it is often due to metavariables
  Unique ?m.174974`
I fixed it in that way but I imagine there is a better way.

Reply:
You can use `← norm_col (ι := Unit)`. The statement of `norm_col` changed in #11536.

Reply:
Thanks!